src/bin/iq.rs

```diff
@@ -241,6 +247,67 @@ fn obtain_iq_client(iq_sub_command: &ArgMatches) -> IQClient {
     return IQClient::new(server.clone(), user, token, stage, application, attempts);
 }
 
+fn print_iq_policy_violations(res: PolicyReportResult, parser: &impl ParseToml) -> () {
+    let policy_violations: Vec<Component> = res
+        .components
+        .clone()
+        .into_iter()
+        .filter(|p| {
+            if p.violations
+                .as_ref()
+                .unwrap_or(&(vec![] as Vec<Violation>))
+                .is_empty()
+            {
+                return false;
+            } else {
+                return true;
+            }
+        })
```

Comment:
This might be too clever, but I wanted to see if I could come up with an approach that didn't `unwrap_or` allocate a new empty vector. This approach instead filters defaults `None` to `false` and then returns the result of `!.is_empty()` on the internal `Vec`. Does this read cleanly to everyone else?

```suggestion
        .filter(|p| p.violations.as_ref().map_or(false, |v| !v.is_empty()))
```

I wish there was an API that would be an API that would do this for us that would make it a bit more ergonomic to use `Option`s in predicates.

Reply:
I had a bit of trouble here tbh, it took me a minute to figure this out. I like your suggestion!